src/query/src/datafusion.rs

```diff
@@ -67,19 +69,54 @@ impl DatafusionQueryEngine {
         }
     }
 
-    fn plan_sql_stmt(&self, stmt: Statement, query_ctx: QueryContextRef) -> Result<LogicalPlan> {
-        let context_provider = DfContextProviderAdapter::new(self.state.clone(), query_ctx);
-        let planner = DfPlanner::new(&context_provider);
-        planner
-            .statement_to_plan(stmt)
-            .map_err(BoxedError::new)
-            .context(QueryPlanSnafu)
+    async fn plan_sql_stmt(
+        &self,
+        stmt: Statement,
+        query_ctx: QueryContextRef,
+    ) -> Result<LogicalPlan> {
+        let session_state = self.state.session_state();
+
+        let df_stmt = (&stmt).try_into().context(SqlSnafu)?;
+
+        let config_options = session_state.config().config_options();
+        let parser_options = ParserOptions {
+            enable_ident_normalization: config_options.sql_parser.enable_ident_normalization,
+            parse_float_as_decimal: config_options.sql_parser.parse_float_as_decimal,
+        };
+
+        let context_provider = DfContextProviderAdapter::try_new(
+            self.state.clone(),
+            session_state,
+            &df_stmt,
+            query_ctx,
+        )
+        .await?;
+        let sql_to_rel = SqlToRel::new_with_options(&context_provider, parser_options);
+
+        let result = sql_to_rel.statement_to_plan(df_stmt).with_context(|_| {
+            let sql = if let Statement::Query(query) = stmt {
+                query.inner.to_string()
+            } else {
+                format!("{stmt:?}")
```

Comment:
I think it's better to format the original parameter (i.e. the `stmt`) here instead of the parsed one `df_stmt`. More user friendly in error reporting.

Reply:
I'm just afraid that always formatting and printing the statement might results in a very long error message which actually isn't user-friendly. Maybe we should implement display for our statement like what sqlparser does.